contracts/PolyLocker.sol

```diff
@@ -13,26 +13,44 @@ pragma solidity 0.5.8;
 
 import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
-import "./PolyLockerStorage.sol";
-import "./proxies/ProxyOwner.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 
 /**
  * @title Contract used to lock POLY corresponds to locked amount user can claim same
  * amount of POLY on Polymesh blockchain
  */
-contract PolyLocker is PolyLockerStorage, ProxyOwner {
+contract PolyLocker is Ownable {
     using SafeMath for uint256;
 
+    // Tracks the total no. of events emitted by the contract.
+    uint256 public noOfeventsEmitted;
+
+    // Address of the token that is locked by the contract. i.e. PolyToken contract address.
+    address public polyToken;
+
+    // Controls if locking Poly is frozen.
+    bool public frozen;
+
+    // Granularity Polymesh blockchain in 10^6 but it's 10^18 on Ethereum.
+    // This is used to truncate 18 decimal places to 6.
+    uint256 constant public TRUNCATE_SCALE = 10 ** 12;
+
+    // Valid address length of Polymesh blockchain.
+    uint256 constant public VALID_ADDRESS_LENGTH = 48;
+
+    uint256 constant internal E18 = uint256(10) ** 18;
+
     // Emit an event when the poly gets lock
     event PolyLocked(uint256 indexed _id, address indexed _holder, string _meshAddress, uint256 _polymeshBalance, uint256 _polyTokenBalance);
     // Emitted when locking is frozen
     event Frozen();
     // Emitted when locking is unfrozen
     event Unfrozen();
 
-    modifier onlyOwner() {
-        require(msg.sender == _upgradeabilityOwner(), "Unauthorized");
-        _;
+
+    constructor(address _polyToken) public {
+        require(_polyToken != address(0), "Invalid address");
+        polyToken = _polyToken;
```

Comment:
We can improve it more in terms of gas perspective. We can store the typecasted variable as an immutable variable so it can live within the bytecode.

```
IERC20 public immutable polyToken;

constructor(address _polyToken) public {
        require(_polyToken != address(0), "Invalid address");
        polyToken = IERC20(_polyToken);
}
```

But to support this we need to update our compiler version.

Reply:
Or we can just hardcode it in the bytecode like `address constant polyToken = 0xDeadBeef`

Reply:
Hmm yeah we can do this as well.
may this will be more suitable 
```
IERC20 constant polyToken = IERC20(0xDeadBeef)
```